File: LEETCODE/MEDIUM/DAY-2/DAY 14/ONE EDIT DISTANCE.py
def isOneEditDistance(s: str, t: str) -> bool:
    m, n = len(s), len(t)
    if m > n:
        return isOneEditDistance(t, s)
    if n - m > 1:
        return False

    i = 0
    while i < m and s[i] == t[i]:
        i += 1

    if i == m:
        return n - m == 1  # Extra char at end

    if m == n:
        i += 1  # Skip one mismatch

    while i < m and s[i] == t[i + (n - m)]:
        i += 1

    return i == m


# 🚀 Example Run
print(isOneEditDistance("ab", "acb"))   # True
print(isOneEditDistance("cab", "ad"))   # False
print(isOneEditDistance("1203", "1213")) # True

File: LEETCODE/MEDIUM/DAY 14/ONE EDIT DISTANCE.java
class Solution {
    public boolean isOneEditDistance(String s, String t) {
        int m = s.length(), n = t.length();
        if (m > n) return isOneEditDistance(t, s);
        if (n - m > 1) return false;

        int i = 0;
        while (i < m && s.charAt(i) == t.charAt(i)) i++;

        if (i == m) return n - m == 1; // Extra char at end

        if (m == n) i++; // Modification case

        while (i < m && s.charAt(i) == t.charAt(i + (n - m))) i++;

        return i == m;
    }

    public static void main(String[] args) {
        Solution sol = new Solution();
        System.out.println(sol.isOneEditDistance("ab", "acb"));   // true
        System.out.println(sol.isOneEditDistance("cab", "ad"));   // false
        System.out.println(sol.isOneEditDistance("1203", "1213")); // true
    }
}

